### PR TITLE
Add ZStream.branchAfter and pipeline

### LINKED ISSUE #5947: ZIO Streams 2.0: Port ZTransducer#branchAfter to ZPipeline

### PR DESCRIPTION
**Pull request by regiskuckaertz**
Closes #5947

### CODE REVIEW

File: streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala [hunk 1]
```diff
@@ -3774,9 +3814,10 @@ class ZStream[-R, +E, +A](val channel: ZChannel[R, Any, Any, Any, E, Chunk[A], A
   /**
    * Threads the stream through the transformation function `f`.
    */
-  final def via[R2, E2, A2](f: ZStream[R, E, A] => ZStream[R2, E2, A2])(implicit
+  final def via[R2 <: R, E2 >: E, A2 >: A, A3](pipeline: ZPipeline[R2, E2, A2, A3])(implicit
```
Comment (vigoo) [inline streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala:3817]: I think we may want to keep the other variant too to help migration, as it also existed on the old `ZStream`

File: streams/shared/src/main/scala/zio/stream/experimental/ZPipeline.scala [hunk 2]
```diff
@@ -397,6 +397,15 @@ object ZPipeline {
...
 
+  /**
+   * Emits the provided chunk before emitting any other value.
+   */
+  def prepend[A](values: Chunk[A]): ZPipeline[Any, Nothing, A, A] =
+    new ZPipeline[Any, Nothing, A, A] {
+      def apply[R, E](stream: ZStream[R, E, A])(implicit trace: ZTraceElement): ZStream[R, E, A] =
+        ZStream.fromChunk(values) ++ stream
+    }
+
```
Comment (vigoo) [inline streams/shared/src/main/scala/zio/stream/experimental/ZPipeline.scala:408]: We could add `branchAfter` that just calls the implemnetation on the stream here too - if I understood @adamgfraser 's recommendation :)

File: streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala (hunk 1 above)
Comment (regiskuckaertz) [reply streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala:3817]: Yeah that makes sense.

File: streams/shared/src/main/scala/zio/stream/experimental/ZPipeline.scala (hunk 2 above)
Comment (regiskuckaertz) [reply streams/shared/src/main/scala/zio/stream/experimental/ZPipeline.scala:408]: Oh, I forgot to do that 🤦

File: streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala (hunk 1 above)
Comment (regiskuckaertz) [reply streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala:3825]: Done

File: streams/shared/src/main/scala/zio/stream/experimental/ZPipeline.scala (hunk 2 above)
Comment (regiskuckaertz) [reply streams/shared/src/main/scala/zio/stream/experimental/ZPipeline.scala:414]: Done

**Review by vigoo — APPROVED** [review]

**Pull request merged**